tests/testthat/test-decision_tree-partykit.R

```diff
@@ -0,0 +1,93 @@
+library(testthat)
+
+test_that("model object", {
+  skip_if_not_installed("partykit")
+  set.seed(1234)
+  exp_f_fit <- partykit::ctree(Surv(time, status) ~ age + ph.ecog, data = lung)
+
+  # formula method
+  cox_spec <- decision_tree() %>%
+    set_mode("censored regression") %>%
+    set_engine("partykit")
+  set.seed(1234)
+  expect_error(
+    f_fit <- fit(cox_spec, Surv(time, status) ~ age + ph.ecog, data = lung),
+    NA
+  )
+
+  # Removing `call` element from comparison
+  f_fit$fit$info$call <- NULL
+  exp_f_fit$info$call <- NULL
+  expect_equal(
+    f_fit$fit,
+    exp_f_fit,
```

Comment:
these two currently differ in their value for `mtry`: 

```
f_fit$fit (`actual`) not equal to `exp_f_fit` (`expected`).

  `actual$info$control$mtry`: 3
  `expected$info$control$mtry`: 2
```
I think partykit takes into account that two columns are needed for the response (via `Surv()`) hence my attempt to change the default value when using `parsnip::cforest_train()` for survival analysis.